tests/fee_estimation/test_fee_estimation_unit_tests.py

```diff
@@ -145,3 +148,36 @@ def test_fee_estimation_inception() -> None:
 
     # Confirm that estimates start after block 4
     assert e1 == [0, 0, 0, 2, 2, 2, 2]
+
+
+def test_init_buckets() -> None:
+    buckets, sorted_buckets = init_buckets()
+    assert len(buckets) == len(sorted_buckets)
+
+
+def test_get_bucket_index_empty_buckets() -> None:
+    sorted_buckets = SortedDict()
+    for rate in [0.5, 1.0, 2.0]:
+        with pytest.raises(RuntimeError):
+            _ = get_bucket_index(sorted_buckets, rate)
+
+
+def test_get_bucket_index_single_entry() -> None:
+    """Test single entry with low, equal and high keys"""
+    sorted_buckets = SortedDict({1.0: 0})
+    print()
+    print(sorted_buckets)
+    for rate, expected_index in ((0.5, 0), (1.0, 0), (1.5, 0)):
+        result_index = get_bucket_index(sorted_buckets, rate)
+        print(rate, expected_index, result_index)
+        assert expected_index == result_index
+
+
+def test_get_bucket_index() -> None:
+    sorted_buckets = SortedDict({1.0: 0, 2.0: 1})
+    print()
+    print(sorted_buckets)
+    for rate, expected_index in ((0.5, 0), (1.0, 0), (1.5, 0), (2.0, 1), (2.1, 1)):
```

Comment:
I think it's risky to have the tests use fee rates that fall exactly on the bucket boundaries. It may make them flaky since floating point comparisons don't (typically) have completely portable behavior.